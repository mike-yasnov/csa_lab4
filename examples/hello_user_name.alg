// Программа hello_user_name - запрашивает имя пользователя и приветствует

function main() {
    print("What is your name?");
    
    // Упрощённая версия - читаем один символ как имя
    var ch = read();
    
    print("Hello, ");
    print(ch);  // Печатаем код символа
    print("!");
}

main(); 