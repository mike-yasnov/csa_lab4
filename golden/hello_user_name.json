{
  "return_code": 0,
  "stdout": "Загрузка программы: /tmp/tempfile/program.bin\nЗагружено 54 инструкций\nЗагрузка данных: /tmp/tempfile/program_data.bin\nЗагружено 70 байт данных\nЗапуск выполнения (максимум 1000000 тактов)...\n\n=== РЕЗУЛЬТАТЫ ВЫПОЛНЕНИЯ ===\nСостояние: halted\nВыполнено инструкций: 73\nЗатрачено тактов: 211\nФинальный PC: 53\n\nВЫВОД ПРОГРАММЫ:\nWhat is your name?\nHello, A!\n\nПрограмма завершена успешно.\n",
  "stderr": "",
  "exec_log": "Cycle 000002: PC=0000 40(000034) Stack[0] TOS=empty\nCycle 000006: PC=0034 43(000001) Stack[0] TOS=empty\nCycle 000008: PC=0001 00(000000) Stack[0] TOS=empty\nCycle 000013: PC=0002 61(000001) Stack[1] TOS=0\nCycle 000015: PC=0003 00(000014) Stack[0] TOS=empty\nCycle 000017: PC=0004 00(000034) Stack[1] TOS=20\nCycle 000021: PC=0005 51(000000) Stack[2] TOS=52\nCycle 000023: PC=0006 00(000034) Stack[0] TOS=empty\nCycle 000027: PC=0007 50(000000) Stack[1] TOS=52\nCycle 000029: PC=0008 00(000038) Stack[1] TOS=20\nCycle 000033: PC=0009 51(000000) Stack[2] TOS=56\nCycle 000035: PC=000A 00(000038) Stack[0] TOS=empty\nCycle 000039: PC=000B 50(000000) Stack[1] TOS=56\nCycle 000041: PC=000C 00(000034) Stack[1] TOS=20\nCycle 000045: PC=000D 50(000000) Stack[2] TOS=52\nCycle 000048: PC=000E 11(000000) Stack[2] TOS=20\nCycle 000050: PC=000F 00(000020) Stack[1] TOS=0\nCycle 000052: PC=0010 00(000001) Stack[2] TOS=32\nCycle 000055: PC=0011 11(000000) Stack[3] TOS=1\nCycle 000058: PC=0012 35(000000) Stack[2] TOS=31\nCycle 000061: PC=0013 42(000027) Stack[1] TOS=0\nCycle 000066: PC=0014 60(000000) Stack[0] TOS=empty\nCycle 000068: PC=0015 02(000000) Stack[1] TOS=65\nCycle 000070: PC=0016 00(000000) Stack[2] TOS=65\nCycle 000073: PC=0017 30(000000) Stack[3] TOS=0\nCycle 000076: PC=0018 42(000027) Stack[2] TOS=0\nCycle 000078: PC=0019 02(000000) Stack[1] TOS=65\nCycle 000080: PC=001A 00(00000A) Stack[2] TOS=65\nCycle 000083: PC=001B 30(000000) Stack[3] TOS=10\nCycle 000086: PC=001C 42(000027) Stack[2] TOS=0\nCycle 000088: PC=001D 00(000038) Stack[1] TOS=65\nCycle 000092: PC=001E 50(000000) Stack[2] TOS=56\nCycle 000096: PC=001F 54(000000) Stack[2] TOS=20\nCycle 000098: PC=0020 00(000038) Stack[0] TOS=empty\nCycle 000102: PC=0021 50(000000) Stack[1] TOS=56\nCycle 000104: PC=0022 00(000001) Stack[1] TOS=20\nCycle 000107: PC=0023 10(000000) Stack[2] TOS=1\nCycle 000109: PC=0024 00(000038) Stack[1] TOS=21\nCycle 000113: PC=0025 51(000000) Stack[2] TOS=56\nCycle 000115: PC=0026 40(00000A) Stack[0] TOS=empty\nCycle 000117: PC=000A 00(000038) Stack[0] TOS=empty\nCycle 000121: PC=000B 50(000000) Stack[1] TOS=56\nCycle 000123: PC=000C 00(000034) Stack[1] TOS=21\nCycle 000127: PC=000D 50(000000) Stack[2] TOS=52\nCycle 000130: PC=000E 11(000000) Stack[2] TOS=20\nCycle 000132: PC=000F 00(000020) Stack[1] TOS=1\nCycle 000134: PC=0010 00(000001) Stack[2] TOS=32\nCycle 000137: PC=0011 11(000000) Stack[3] TOS=1\nCycle 000140: PC=0012 35(000000) Stack[2] TOS=31\nCycle 000143: PC=0013 42(000027) Stack[1] TOS=0\nCycle 000148: PC=0014 60(000000) Stack[0] TOS=empty\nCycle 000150: PC=0015 02(000000) Stack[1] TOS=10\nCycle 000152: PC=0016 00(000000) Stack[2] TOS=10\nCycle 000155: PC=0017 30(000000) Stack[3] TOS=0\nCycle 000158: PC=0018 42(000027) Stack[2] TOS=0\nCycle 000160: PC=0019 02(000000) Stack[1] TOS=10\nCycle 000162: PC=001A 00(00000A) Stack[2] TOS=10\nCycle 000165: PC=001B 30(000000) Stack[3] TOS=10\nCycle 000168: PC=001C 42(000027) Stack[2] TOS=1\nCycle 000170: PC=0027 00(000038) Stack[1] TOS=10\nCycle 000174: PC=0028 50(000000) Stack[2] TOS=56\nCycle 000176: PC=0029 00(000000) Stack[2] TOS=21\nCycle 000178: PC=002A 03(000000) Stack[3] TOS=0\nCycle 000182: PC=002B 54(000000) Stack[3] TOS=21\nCycle 000184: PC=002C 00(00003C) Stack[1] TOS=10\nCycle 000189: PC=002D 61(000001) Stack[2] TOS=60\nCycle 000191: PC=002E 00(000034) Stack[1] TOS=10\nCycle 000195: PC=002F 50(000000) Stack[2] TOS=52\nCycle 000200: PC=0030 61(000001) Stack[2] TOS=20\nCycle 000202: PC=0031 00(000044) Stack[1] TOS=10\nCycle 000207: PC=0032 61(000001) Stack[2] TOS=68\nCycle 000210: PC=0033 44(000000) Stack[1] TOS=10\nCycle 000211: PC=0035 70(000000) Stack[1] TOS=10",
  "debug_listing": "INSTRUCTION MEMORY:\nAddress - Hex Code - Mnemonic\n----------------------------------------\n0000 - 40340000 - JMP 52\n0002 - 00000000 - PUSH\n0004 - 61010000 - OUT 1\n0006 - 00140000 - PUSH 20\n0008 - 00340000 - PUSH 52\n000A - 51000000 - STORE\n000B - 00340000 - PUSH 52\n000D - 50000000 - LOAD\n000E - 00380000 - PUSH 56\n0010 - 51000000 - STORE\n0011 - 00380000 - PUSH 56\n0013 - 50000000 - LOAD\n0014 - 00340000 - PUSH 52\n0016 - 50000000 - LOAD\n0017 - 11000000 - SUB\n0018 - 00200000 - PUSH 32\n001A - 00010000 - PUSH 1\n001C - 11000000 - SUB\n001D - 35000000 - GE\n001E - 42270000 - JNZ 39\n0020 - 60000000 - IN\n0022 - 02000000 - DUP\n0023 - 00000000 - PUSH\n0025 - 30000000 - EQ\n0026 - 42270000 - JNZ 39\n0028 - 02000000 - DUP\n0029 - 000A0000 - PUSH 10\n002B - 30000000 - EQ\n002C - 42270000 - JNZ 39\n002E - 00380000 - PUSH 56\n0030 - 50000000 - LOAD\n0031 - 54000000 - STOREB\n0032 - 00380000 - PUSH 56\n0034 - 50000000 - LOAD\n0035 - 00010000 - PUSH 1\n0037 - 10000000 - ADD\n0038 - 00380000 - PUSH 56\n003A - 51000000 - STORE\n003B - 400A0000 - JMP 10\n003D - 00380000 - PUSH 56\n003F - 50000000 - LOAD\n0040 - 00000000 - PUSH\n0042 - 03000000 - SWAP\n0043 - 54000000 - STOREB\n0044 - 003C0000 - PUSH 60\n0046 - 61010000 - OUT 1\n0048 - 00340000 - PUSH 52\n004A - 50000000 - LOAD\n004B - 61010000 - OUT 1\n004D - 00440000 - PUSH 68\n004F - 61010000 - OUT 1\n0051 - 44000000 - RET\n0052 - 43010000 - CALL 1\n0054 - 70000000 - HALT\n\nDATA MEMORY:\nAddress - Hex Dump - ASCII\n----------------------------------------\n0000 - 57 68 61 74 20 69 73 20 79 6F 75 72 20 6E 61 6D  - What is your nam\n0010 - 65 3F 0A 00 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F  - e?..____________\n0020 - 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F 5F  - ________________\n0030 - 5F 5F 5F 5F 00 00 00 00 00 00 00 00 48 65 6C 6C  - ____........Hell\n0040 - 6F 2C 20 00 21 00                                - o, .!.\n"
}